pkgs/applications/networking/browsers/brave/default.nix

```
@@ -41,63 +40,61 @@
 , xorg
 , zlib
 , xdg-utils
-, wrapGAppsHook
+, snappy
+, libopus
+
+# command line arguments which are always set e.g "--disable-gpu"
 , commandLineArgs ? ""
+
+# Necessary for USB audio devices.
+, pulseSupport ? true, libpulseaudio ? null
```

```suggestion
, pulseSupport ? true, libpulseaudio
```
That is never null unless you overwrite it to null which is unsupported.

---

I have followed the lead of `google-chrome` and `chromium` where both are set up like that, so I assumed it's a desired pattern.  Is there any reason not to follow that?

---

Yes, if they have the same code they are also doing it in a legacy way.

> Is there any reason not to follow that?

It hides bugs. The following would be valid and always null.

```suggestion
, pulseSupport ? true, bogusPackages123 ? null
```

---

Okay, I have removed the defaults. I have also changed the default for pulse/libva support to `stdenv.isLinux`, since they only really make sense on loonices.